typing/subst.mli

```diff
@@ -29,22 +28,36 @@ type t
    Indeed, non-variable node of a type are duplicated, with their
    levels set to generic level.  That way, the resulting type is
    well-formed (decreasing levels), even if the original one was not.
-*)
 
-val identity: t
+   In the presence of local substitutions for module types, a substitution for a
+   type expression may fail to produce a well-formed type. In order to confine
+   this issue to local substitution, the type of substitutions is split into a
+   standard and local variant. Only local substitution may expand a module type
+   path into a generic module type. *)
+
+(** Type familly for substitutions *)
+type 'a s
+
+type t = [`safe] s
+(** Standard substitution*)
+
+type local = [`local] s
+(** Local substitution *)
 
-val add_type: Ident.t -> Path.t -> t -> t
-val add_type_path: Path.t -> Path.t -> t -> t
+val identity: 'a s
+val local: t -> local
+
+val add_type: Ident.t -> Path.t -> 'a s -> 'a s
+val add_type_path: Path.t -> Path.t -> 'a s -> 'a s
 val add_type_function:
-  Path.t -> params:type_expr list -> body:type_expr -> t -> t
-val add_module: Ident.t -> Path.t -> t -> t
-val add_module_path: Path.t -> Path.t -> t -> t
-val add_modtype: Ident.t -> module_type -> t -> t
-val add_modtype_path: Path.t -> module_type -> t -> t
+  Path.t -> params:type_expr list -> body:type_expr -> 'a s -> 'a s
+val add_module: Ident.t -> Path.t -> 'a s -> 'a s
+val add_module_path: Path.t -> Path.t -> 'a s -> 'a s
+val add_modtype_id_to_path: Ident.t -> Path.t -> 'a s -> 'a s
```

Comment:
Naming proposal: `add_modtype : Ident.t -> Path.t -> 'a s -> 'a s`. It is only safe to expand a module type to a path, so it makes to name this as the canonical substitution operation for module types, as we are doing for modules above.

Reply:
I have aligned those function names to the one used for modules.